### DISCUSSION TITLE
Weblate on ECS with Volumes and SES

### DISCUSSION THREAD

**Discussion by AKIvan** [WeblateOrg/weblate#9843] [Q&A] [answered]
Hi all, 
I'm running the weblate on AWS ECS. And i have two problems. 

1. I have defined the env variables for the SES or following the Email setup i have defined the WEBLATE_EMAIL_HOST, WEBLATE_EMAIL_HOST_USER ,WEBLATE_EMAIL_HOST_PASSWORD. Using the same credentials with Boto3 i can send emails. So im guessing that AWS SES is configured properly. Anything im missing here, as im not getting any e-mails from weblate, like when i have to verify the new e-mails or passwords etc. 

2. Is the volume part where i have setup the volume for weblate data - this is more of the need for the backup of the database that is happening every day and i need to have access to this file. The problem im having it s to get to that data from ECS. So my question is, is there a way to store the daily backup data to an S3 bucket ?

The whole project is deployed via CDK.

**Comment by AKIvan** [comment]
Anyone on this one at least to have some info ?

**Comment by michael-smt** [comment]
Hello there fellow Weblate on AWS ECS runner.

1. Boto probably sends via the SES API. Weblate uses SMTP. Check out the [Requirements to send email over SMTP](https://docs.aws.amazon.com/ses/latest/dg/send-email-smtp.html#send-email-smtp-requirements) chapter in the AWS SES docs.
2. Not quite sure what you mean. Weblate provides BorgBackup [built-in](https://docs.weblate.org/en/latest/admin/backup.html), which does [not appear to support S3](https://github.com/borgbackup/borg/issues/102), only SFTP.
    In our environment we use the native RDS backup for the DB. For files we have a AWS EFS mounted at `/app/data` and use EFS's native backup method.

**Reply by AKIvan** [reply]
1. I thought setting up those variables and credentials are the only place to configure the SMTP. Ill back to this once i dig deeper. 
2. I'm using the weblate/weblate image, i guess it comes with BorgBackup. I mount the EFS to same path as yours `/app/data`. 
Now when i go to Weblate Web interface and to the manage part and than Backups i "Add backup service" to be /app/data. Now i got backup failed with `/app/data is not a valid repository. Check repo config.` but on the other hand i see files in the EFS. I see the new `database.sql` and other files along. 
Now why im getting this error message but on the other hand the backup is there?
If I understand correctly BorgBackup is one of more methods of doing backup and there is a regular one or default one that is storing or making weblate a backup in this directory in the `/app/data/backups/` ?

**Reply by nijel** [reply, marked as answer]
1. SES should be usable directly via https://github.com/django-ses/django-ses, but it is currently not integrated in the Weblate Docker container.
2. `/app/data/backups` is used for internal backups (see https://docs.weblate.org/en/latest/admin/config.html#std-setting-DATA_DIR), please choose a different location (I've added validation for this in 49e78a0e6ea0508130294ec62650e99a0fe69df5).

**Reply by AKIvan** [reply]
OK so for the backup as i understand, the local backups or what i would call them Local Backups or Internal backups, that is happening under `/app/data/backups` will be executed and will happen even if i dont have anything configured in the "Manage" > "Backups" part.
And if i would like to have the Borg backups or procedures i need to configure it in the   "Manage" > "Backups" using different paths.

**Reply by nijel** [reply]
`/app/data/backups` is used for data included in the Borg backups like database or configuration dumps.

**Comment by AKIvan** [comment]
Thank you for the support, it will be good to have SES in the container somehow, already ready, and just use the env variables.

**Reply by nijel** [reply]
It should be easy to integrate, contributions are welcome:

* add package to https://github.com/WeblateOrg/docker/blob/main/requirements.txt
* add env parsing to https://github.com/WeblateOrg/weblate/blob/main/weblate/settings_docker.py
